ethpm_types/contract_type.py

```diff
@@ -129,6 +129,56 @@ def extract_sourcemap_item(expanded_row, item_idx, previous_val=None):
             yield item
 
 
+class ABIList(list):
+    """
+    Adds selection by name, selector and keccak(selector).
+    """
+
+    def __init__(
+        self,
+        iterable=(),
+        *,
+        selector_id_size=32,
+        selector_hash_fn: Optional[Callable[[str], bytes]] = None,
+    ):
+        self._selector_id_size = selector_id_size
+        self._selector_hash_fn = selector_hash_fn
+        super().__init__(iterable)
+
+    def __getitem__(self, item: Union[str, bytes, MethodABI, EventABI]):  # type: ignore
+        try:
+            # selector
+            if isinstance(item, str) and "(" in item:
+                return next(abi for abi in self if abi.selector == item)
+            # name, could be ambiguous
+            elif isinstance(item, str):
+                return next(abi for abi in self if abi.name == item)
+            # hashed selector, like log.topics[0] or tx.data
+            elif isinstance(item, bytes) and self._selector_hash_fn:
+                return next(
+                    abi
+                    for abi in self
+                    if self._selector_hash_fn(abi.selector)[: self._selector_id_size]
+                    == item[: self._selector_id_size]
+                )
+            elif isinstance(item, (MethodABI, EventABI)):
+                return next(abi for abi in self if abi.selector == item.selector)
+        except StopIteration:
+            raise KeyError(item)
+
+        # handle int, slice
+        return super().__getitem__(item)  # type: ignore
+
+    def __contains__(self, item: Union[str, bytes]) -> bool:  # type: ignore
```

Comment:
Is there any harm in also supporting `MethodABI` and `EventABI` here?

Reply:
i thought about wrapping the whole ABI in ABIList, and the problem is the different type have a different selector length (e.g. 4 bytes for functions, 32 bytes for event topics on ethereum)